src/utils/dump.h

```diff
@@ -0,0 +1,51 @@
+/*
+  Copyright (c) [2023-2024] [Sparq Network]
+
+  This software is distributed under the MIT License.
+  See the LICENSE.txt file in the project root for more information.
+*/
+
+#ifndef DUMP_H
+#define DUMP_H
+
+#include <vector>
+
+/**
+ * Abstraction of the dumpable object.
+ */
+class Dumpable {
+public:
+  /**
+   * Pure virtual function to be implemented.
+   * The function should dump implemented by the methods that are dumpable.
+   */
+  virtual void dump() = 0;
+};
+
+/**
+ * Dumpable management.
+ * Used to store dumpable objects in memory.
+ */
+class DumpManager {
+private:
+  std::vector<Dumpable*> dumpable_;
```

Comment:
Why not `std::vector<std::reference_wrapper<Dumpable>>`? I think that Dumpable cannot be null in this context.

Reply:
ok